Se recomienda estudiar la realización de las siguientes instrucciones en python:
1. EDA
importar datasets

*   averiguar tamaño del dataset, información, resumen y ver la cabecera.
*   sustituir valores en un dataset, saber si hay nulos.
*   representar histogramas de los datos
*   eliminar atributos del dataset
*   representar la matriz de correlacion
2. Tarea de regresion
*   normalizar los datos
*   separar los datos en train y test
*   como realizar una regresión lineal
*   como realizar un árbol de decisión
*   como obtener las metricas para un problema de regresión

# **1. Análisis descriptivo de los datos**


### 1.1 Cargar el dataset "House Sales in King County, USA" 
Puedes descargarlo aquí: https://www.kaggle.com/datasets/harlfoxem/housesalesprediction/data

In [1]:
import pandas as pd

dataset = pd.read_csv('kc_house_data.csv', delimiter=',')

### 1.2 ¿Cuántas instancias tiene el dataset?

In [2]:
dataset.shape

(21613, 21)

In [3]:
print(dataset.head(5))

           id             date     price  bedrooms  bathrooms  sqft_living  \
0  7129300520  20141013T000000  221900.0         3       1.00         1180   
1  6414100192  20141209T000000  538000.0         3       2.25         2570   
2  5631500400  20150225T000000  180000.0         2       1.00          770   
3  2487200875  20141209T000000  604000.0         4       3.00         1960   
4  1954400510  20150218T000000  510000.0         3       2.00         1680   

   sqft_lot  floors  waterfront  view  ...  grade  sqft_above  sqft_basement  \
0      5650     1.0           0     0  ...      7        1180              0   
1      7242     2.0           0     0  ...      7        2170            400   
2     10000     1.0           0     0  ...      6         770              0   
3      5000     1.0           0     0  ...      7        1050            910   
4      8080     1.0           0     0  ...      8        1680              0   

   yr_built  yr_renovated  zipcode      lat     lo

El dataset tiene **21613** instancias. Cada instancia se compone de **21** variables:
- id: Identificador único para cada casa vendida.
-date: Fecha en la que se realizó la venta de la propiedad.
- price: Precio de venta de la vivienda (variable objetivo).
- bedrooms: Número de habitaciones de la casa.
- bathrooms: Número de baños, donde .5 indica un baño sin ducha.
- sqft_living: Pies cuadrados de espacio habitable interior.
- sqft_lot: Pies cuadrados de la superficie total del terreno.
- floors: Número de pisos o niveles que tiene la casa.
- waterfront: Variable binaria que indica si la casa tiene vistas al mar/agua.
- view: Índice de 0 a 4 sobre qué tan buena es la vista desde la propiedad.
- condition: Índice de 1 a 5 que mide la condición estructural de la casa.
- grade: Índice de 1 a 13 que califica el nivel de construcción y diseño.
- sqft_above: Pies cuadrados de espacio interior que están por encima del nivel del suelo.
- sqft_basement: Pies cuadrados de espacio interior que están por debajo del nivel del suelo.
- yr_built: Año en el que se construyó originalmente la vivienda.
- yr_renovated: Año de la última remodelación (0 si nunca se ha renovado).
- zipcode: Código postal del área donde se ubica la casa.
- lat: Coordenada de latitud geográfica.
- long: Coordenada de longitud geográfica.
- sqft_living15: Pies cuadrados promedio del espacio habitable de los 15 vecinos más cercanos.
- sqft_lot15: Pies cuadrados promedio del terreno de los 15 vecinos más cercanos.

### 1.3 ¿Cuál es el tipo de datos de cada una de las columnas?

**Nota**: CUIDADO CON TENER ATRIBUTOS COMO OBJECT

### 1.4 ¿Cuántas columnas categóricas hay? ¿y cuántas continuas?

### 1.5 ¿Existen valores nulos en el dataset?

### 1.6 ¿Cuál es la variable respuesta?¿de qué tipo es?

### 1.7 ¿Cómo es la distribución de las variables: bedrooms, bathrooms, floors y condition?

### 1.8 ¿Cómo detectarías valores anómalos? Invéntate un ejemplo de valor anómalo y cómo lo detectarías

### 1.9 Cuenta la cantidad de casas que tienen cero bathrooms

### 1.10 Supón que 0 bathrooms es un error. Sustituye el valor 0 por nan

### 1.11 Elimina la variable date y waterfront

### 1.12 ¿Cómo son las correlaciones entre las variables del dataset?

### 1.13 ¿Qué tres variables son las más correlacionadas con la variable objetivo?

# **2. Tarea de regresión**

### 2.1 Escalado de datos

### 2.2 Z-Score



### 2.3 Separar datos de entrenamiento y test

### 2.4 Entrena el modelo con los datos de entrenamiento

### 2.5 Creamos un objeto de Regresion lineal

### 2.6 Realiza las predicciones

### 2.7 Evaluar el modelo
**Nota**: Usa más de una métrica para tener una mejor comprensión del rendimiento del modelo

### 2.8 Entrenar un segundo modelo que haga uso de una regresión lineal múltiple con todas las variables del dataset

Baseline RMSE of model: 0.026960328127632053


# **3. Regresión con arbol de decision**

### 3.1 Entrena un modelo de árboles de decisión

### 3.2 Cálculo de errores